docs/Makefile

```diff
@@ -5,7 +5,11 @@
 # Treat warnings as errors.... hierarchy.rst is just a dump of notes ATM
 # SPHINXOPTS    = -W
 SPHINXOPTS    =
-SPHINXBUILD   = sphinx-build
+# sphinx-build might be coming from system-wide install, and would
+# use /usr/bin/python3 instead of /usr/bin/env python3, so it would not
+# find our module installed under virtualenv
+SPHINXCMD     = $(shell which sphinx-build)
+SPHINXBUILD   = python3 $(SPHINXCMD)
```

Comment:
Hmm, I'm not sure this is a good idea.  By hard coding python3, you're assuming the system has a python3 executable.  Instead can't a user rely on the `[devel-docs]` target so that sphinx-build is present in the virtualenv regardless of their system?

Reply:
yeah, probably should have been just `python`.  But without this change, I cannot build docs in a virtualenv with sphinx coming from the system-wide installation, sphinx then fails to find reproman module.

Reply:
So why not make a virtualenv that includes sphinx rather than use the system-wide one?  That way, it works for you while not breaking it for people that don't share your system setup.

Reply:
I typically (and how our CONTRIBUTING.md says ATM) virtualenv's are initiated with `--system-site-packages`.  That makes their initiation faster, verifies correct operation with the system-wide packages (and that possibly we need to boost up some versioned depends if not).  We test on travis without relying on system wide installed versions.  Altogether that broadens the coverage.

> ... it works for you while not breaking it for people ...
Things shouldn't break on my or anyone else's system, if they break -- we should have them fixed.

Reply:
> Things shouldn't break on my or anyone else's system, if they break -- we should have them fixed.

It breaks when they don't have a python3 executable.

Reply:
once again, I think it would be better to just change it to `python`.  Having said that, do you know a distribution where having any version of python3 installed would result in no `python3` executable?  on debian it seems to be not the case. installing `python3-attr` resulted in installing `python3-minimal` which provides python3.
Please let me know if I should submit a PR for it , or you will include it in the follow up (to address other comments)

Reply:
> Please let me know if I should submit a PR for it , or you will include it in the follow up (to address other comments)

I don't plan on submitting a follow-up PR.  On master I've already addressed the two minor points  that I intended to include with this PR before the merge: 4739fea9f, 1da8a366d